odoo/cli/shell.py

```diff
@@ -78,6 +78,11 @@ def console(self, local_vars):
             else:
                 shells_to_try = self.supported_shells
 
+            if pythonstartup := config.options.get('shell_file') or os.environ.get('PYTHONSTARTUP'):
+                if not os.path.exists(pythonstartup):
+                    pythonstartup = None
+                os.environ['PYTHONSTARTUP'] = pythonstartup
```

Comment:
I find it strange to set an environment variable all times. It looks to me that is required only for bpython and other shells have an API to run a script at startup.

Reply:
I can pass the variable around...
But let's consider that `os.environ` is NOT the environment, it's a copy of `posix.environ` made at start time. Just a piece of global state.
So if you exit the shell, the environment is unmodified.

Reply:
Yes, it is the environment seen by the current process. As you say, it is a global state and I don't think we need to change it.